bgpd/bgp_packet.c

```diff
@@ -1430,6 +1430,19 @@ static int bgp_keepalive_receive(struct peer *peer, bgp_size_t size)
 
 	bgp_update_implicit_eors(peer);
 
+	peer->rtt = sockopt_tcp_rtt(peer->fd);
+
+	/* If the peer's RTT is higher than expected, shutdown
+	 * the peer automatically.
+	 */
+	if (CHECK_FLAG(peer->flags, PEER_FLAG_RTT_SHUTDOWN)
+	    && peer->rtt > peer->rtt_expected) {
```

Comment:
While this is a good tool but I think it is going to be an issue if we do not provide control a latency for only few packets. What if everything is good but just that one keepalive cross RTT treshold set, will it not be too harsh to shutdown peer? When do we bring it back up agian then?

Reply:
Brink back manually. It depends on the _user_ to apply this or not. We actually hit this _latency_ bug and I had to shut down the peer, fix latencies and bring it back.

If talking about the number of keepalives to shut down, I mean if 3 times or 5 times the latency is higher than X, then shutdown would work too.

Reply:
I can see the value of checking the latency across ```x number of keeps...``` Seems like this might be configurable, as well, though ... I'm okay with this going in "as is," but the additional control would be useful.

Reply:
I can add that.

Reply:
@riw777 @Spantik added.